### PR TITLE
fix: set store before creating proposal

### PR DESCRIPTION
**Pull request by r4mmer**
### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by andreabadesso — APPROVED** [review]

File: src/controllers/wallet/p2sh/tx-proposal.controller.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ async function buildTxProposal(req, res) {
     }
   }
   try {
+    storage.setStore(req.wallet.store);
```
Comment (pedroferreira1) [inline src/controllers/wallet/p2sh/tx-proposal.controller.js:37]: I would just add a comment here saying that this should be in the lib and that we should move it there as soon as we have a more complete method to do it in the wallet class.

File: __tests__/integration/multisig.test.js [hunk 2]
```diff
@@ -267,4 +267,35 @@ describe('send tx (HTR)', () => {
...
+      const decoded = output.parseScript(network);
+      if (decoded.address.base58 === burnAddress) {
+        // This is the intended output
+        expect(decoded.getType()).toBe('p2pkh');
+        expect(output.value).toBe(10);
+        continue;
+      }
+
+      // this is a change address and should be p2sh
+      expect(decoded.getType()).toBe('p2sh');
```
Comment (pedroferreira1) [inline __tests__/integration/multisig.test.js:298]: Not only p2sh, I would test that it belongs to the wallet sending the transaction as well

File: __tests__/integration/multisig.test.js [hunk 3]
```diff
@@ -267,4 +267,35 @@ describe('send tx (HTR)', () => {
     expect(response.body.success).toBe(false);
     expect(response.body.error).toBeDefined();
   });
+
+  it('should mount the transaction with the correct change outputs', async () => {
```
Comment (pedroferreira1) [inline __tests__/integration/multisig.test.js:271]: Do we have a test that we use `tx-proposal` with the `change_address` parameter filled?

File: __tests__/integration/multisig.test.js (hunk 2 above)
Comment (r4mmer) [reply __tests__/integration/multisig.test.js:298]: The underlying wallet-lib method already makes this check, should we also make this check?

File: __tests__/integration/multisig.test.js (hunk 3 above)
Comment (r4mmer) [reply __tests__/integration/multisig.test.js:271]: Yes, and we test when the change address belongs to the wallet and that it fails when it does not belong to the wallet.

File: __tests__/integration/multisig.test.js (hunk 2 above)
Comment (r4mmer) [reply __tests__/integration/multisig.test.js:298]: Done

File: src/controllers/wallet/p2sh/tx-proposal.controller.js (hunk 1 above)
Comment (r4mmer) [reply src/controllers/wallet/p2sh/tx-proposal.controller.js:38]: Done

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**